backend/src/test/java/com/google/blackswan/mock/SimpleAnomalyGeneratorTest.java

```diff
@@ -0,0 +1,120 @@
+  private SimpleAnomalyGenerator anomalyGenerator; 
+      
+
+  @Before
+  public void setUp() throws Exception {
+    helper.setUp();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Test
+  public void getAnomalies_returnsListOfAnomaliesWithSizeOne() {
+    anomalyGenerator = SimpleAnomalyGenerator.createGeneratorWithString(
+      inputForAnomalyGenerator(), SET_THRESHOLD_HIGH, SET_DATAPOINTS
+    );
+    // Avg = 70, only 1 data point exceeds 76. 
+    ImmutableMap<Timestamp, MetricValue> expectedDataPoints = ImmutableMap.of(
+      new Timestamp("2019-07-28"), new MetricValue(59),
+      new Timestamp("2019-08-04"), new MetricValue(75), 
+      new Timestamp("2019-08-11"), new MetricValue(79), 
+      new Timestamp("2019-08-18"), new MetricValue(67)
+    );
+    Anomaly expectedAnomaly = new Anomaly(
+      new Timestamp("2019-08-11"), METRIC_NAME, DIMENSION_NAME, expectedDataPoints
+    );
+
+    List<Anomaly> generatedAnomalies = anomalyGenerator.getAnomalies();
+
+    assertEquals(generatedAnomalies.size(), 1);
+    assertEquals(generatedAnomalies.get(0), expectedAnomaly);
+  }
+
+  @Test
+  public void getAnomalies_returnsListOfAnomaliesWithSizeTwo() {
+    anomalyGenerator = SimpleAnomalyGenerator.createGeneratorWithString(
+      inputForAnomalyGenerator(), SET_THRESHOLD_LOW, SET_DATAPOINTS
+    );
+    // Avg = 70, only 2 data points exceed 74.
+    ImmutableMap<Timestamp, MetricValue> expectedDataPoints1 = ImmutableMap.of(
+      new Timestamp("2019-07-21"), new MetricValue(73),
+      new Timestamp("2019-07-28"), new MetricValue(59),
+      new Timestamp("2019-08-04"), new MetricValue(75), 
+      new Timestamp("2019-08-11"), new MetricValue(79), 
+      new Timestamp("2019-08-18"), new MetricValue(67)
+    );
+    Anomaly expectedAnomaly1 = new Anomaly(
+      new Timestamp("2019-08-04"), METRIC_NAME, DIMENSION_NAME, expectedDataPoints1
+    );
+    ImmutableMap<Timestamp, MetricValue> expectedDataPoints2 = ImmutableMap.of(
+      new Timestamp("2019-07-28"), new MetricValue(59),
+      new Timestamp("2019-08-04"), new MetricValue(75), 
+      new Timestamp("2019-08-11"), new MetricValue(79), 
+      new Timestamp("2019-08-18"), new MetricValue(67)
+    );
+    Anomaly expectedAnomaly2 = new Anomaly(
+      new Timestamp("2019-08-11"), METRIC_NAME, DIMENSION_NAME, expectedDataPoints2
+    );
+
+    List<Anomaly> generatedAnomalies = anomalyGenerator.getAnomalies();
+
+    assertEquals(generatedAnomalies.size(), 2);
+    assertEquals(generatedAnomalies.get(0), expectedAnomaly1);
+    assertEquals(generatedAnomalies.get(1), expectedAnomaly2);
+  }
+
+  private static String inputForAnomalyGenerator() {
+    StringBuilder str = new StringBuilder("");
+    for (Map.Entry<String, Integer> entry : SAMPLE_DATA.entrySet()) {
+      str.append(entry.getKey() + "," + entry.getValue().toString() + "\n");
+    }
+    return str.toString();
+  }
```

Comment:
Alternatively, since the exact code is being written for both tests, you can also have test helper classes/files which are specifically only for test purposes.

ie instead of private static String inputForAnomalyGenerator(), you can have the static String inputForAnomalyGenerator(map_structure), and reuse it in both tests.

Reply:
yep, i was considering creating a helper class/file before, but I wasn't too sure if it was too much for just one function. 

but i think it is definitely good practice, since the two functions are exactly the same!